src/v/cluster/members_backend.cc

```diff
@@ -534,14 +535,26 @@ ss::future<> members_backend::reallocate_replica_set(
   members_backend::partition_reallocation& meta) {
     auto current_assignment = _topics.local().get_partition_assignment(
       meta.ntp);
-    // topic was deleted, we are done with reallocation
+    const auto is_reconfiguring = _topics.local().is_update_in_progress(
+      meta.ntp);
+    //  topic was deleted, we are done with reallocation
     if (!current_assignment) {
         meta.state = reallocation_state::finished;
         co_return;
     }
 
     switch (meta.state) {
     case reallocation_state::initial: {
+        if (is_reconfiguring) {
+            vlog(
+              clusterlog.info,
+              "[ntp: {}, {} -> {}] partition is in the middle of "
+              "reconfiguration, requesting it to cancel",
+              meta.ntp,
+              meta.initial_assignment,
+              get_new_replicas(meta));
+            meta.state = reallocation_state::request_cancel;
```

Comment:
Are we going to cancel all movements of chosen ntps? I.e. even if we are adding a node, are we going to choose some ntps to move to it and cancel any ongoing reconfigurations related to them?

Reply:
I've changed this completely